pyW215/pyW215.py

```diff
@@ -124,29 +140,30 @@ def SOAPAction(self, Action, responseElement, params = ""):
         :return: Text enclosed in responseElement brackets
         """
         # Authenticate client
-        auth = self.auth()
+        if self.authenticated is None:
+            self.authenticated = self.auth()
+        auth = self.authenticated
+        #If not legacy protocol, ensure auth() is called for every call
+        if not self.use_legacy_protocol:
```

Comment:
Is the logic right here? Don't we want to reuse self.authentication for plugs with updated firmware and have self.authentication = None for legacy ones?

Reply:
I believe my code is correct for that part.
For firmware 2+, no legacy protocol, we want to authenticate for every call (as it is right now in master)
For firmware 1.25, use legacy protocol, only authenticate once.
(the thing I am not sure though, is if it will work for people on 1.24...)

Reply:
Aha.. now I get it. I was convinced you were running firmware v2.25 and not v1.25. Your code makes perfect sense.